marketplace/src/lib.rs

```diff
@@ -83,679 +76,503 @@ pub mod pallet {
-				.ok_or(Error::<T>::MarketplaceNotFound)?;
-
-			let commission_fee = market.commission_fee;
-			let mut price = sale.price;
-
-			// KeepAlive because they need to be able to use the NFT later on
-			if commission_fee != 0 {
-				let tmp = 100u8.checked_div(commission_fee).ok_or(Error::<T>::InternalMathError)?;
-
-				let fee = price.checked_div(&(tmp.into())).ok_or(Error::<T>::InternalMathError)?;
-
-				price = price.checked_sub(&fee).ok_or(Error::<T>::InternalMathError)?;
-
-				T::Currency::transfer(&caller, &market.owner, fee, KeepAlive)?;
-			}
+	/// How much does it cost to create a marketplace.
+	#[pallet::storage]
+	#[pallet::getter(fn marketplace_mint_fee)]
+	pub type MarketplaceMintFee<T: Config> =
+		StorageValue<_, BalanceOf<T>, ValueQuery, T::InitialMintFee>;
 
-			T::Currency::transfer(&caller, &sale.account_id, price, KeepAlive)?;
+	/// Counter for marketplace ids.
+	#[pallet::storage]
+	#[pallet::getter(fn next_marketplace_id)]
+	pub type NextMarketplaceId<T: Config> = StorageValue<_, MarketplaceId, ValueQuery>;
 
-			T::NFTExt::set_listed_for_sale(nft_id, false)?;
-			T::NFTExt::set_owner(nft_id, &caller)?;
+	/// Data related to marketplaces
+	#[pallet::storage]
+	#[pallet::getter(fn marketplaces)]
+	pub type Marketplaces<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		MarketplaceId,
+		MarketplaceData<T::AccountId, BalanceOf<T>, T::AccountSizeLimit, T::OffchainDataLimit>,
+		OptionQuery,
+	>;
 
-			NFTsForSale::<T>::remove(nft_id);
+	/// Data related to sales
+	#[pallet::storage]
+	#[pallet::getter(fn nfts_for_sale)]
+	pub type NftsForSale<T: Config> =
+		StorageMap<_, Blake2_128Concat, NFTId, Sale<T::AccountId, BalanceOf<T>>, OptionQuery>;
 
-			let event = Event::NFTSold { nft_id, owner: caller };
-			Self::deposit_event(event);
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Marketplace created
+		MarketplaceCreated {
+			marketplace_id: MarketplaceId,
+			owner: T::AccountId,
+			kind: MarketplaceType,
+			commission_fee: Option<MarketplaceFee<BalanceOf<T>>>,
+			listing_fee: Option<MarketplaceFee<BalanceOf<T>>>,
+			account_list: Option<BoundedVec<T::AccountId, T::AccountSizeLimit>>,
+			offchain_data: Option<U8BoundedVec<T::OffchainDataLimit>>,
+		},
+		/// Marketplace owner set
+		MarketplaceOwnerSet { marketplace_id: MarketplaceId, owner: T::AccountId },
+		/// Marketplace kind set
+		MarketplaceKindSet { marketplace_id: MarketplaceId, kind: MarketplaceType },
+		/// Marketplace config set
+		MarketplaceConfigSet {
+			marketplace_id: MarketplaceId,
+			commission_fee: ConfigOp<MarketplaceFee<BalanceOf<T>>>,
+			listing_fee: ConfigOp<MarketplaceFee<BalanceOf<T>>>,
+			account_list: ConfigOp<BoundedVec<T::AccountId, T::AccountSizeLimit>>,
+			offchain_data: ConfigOp<U8BoundedVec<T::OffchainDataLimit>>,
+		},
+		/// Marketplace mint fee set
+		MarketplaceMintFeeSet { fee: BalanceOf<T> },
+		/// Nft listed
+		NftListed {
```

Comment:
In the events capitalize the word NFT. Only the storage names need to use the following scheme "Nfts"